pkg/ebpf/probes/probes.go

```diff
@@ -2,178 +2,503 @@ package probes
+		DebugfsCreateFile:          &traceProbe{eventName: "debugfs_create_file", probeType: kprobe, programName: "trace_debugfs_create_file"},
+		DebugfsCreateDir:           &traceProbe{eventName: "debugfs_create_dir", probeType: kprobe, programName: "trace_debugfs_create_dir"},
+		DeviceAdd:                  &traceProbe{eventName: "device_add", probeType: kprobe, programName: "trace_device_add"},
+		RegisterChrdev:             &traceProbe{eventName: "__register_chrdev", probeType: kprobe, programName: "trace___register_chrdev"},
+		RegisterChrdevRet:          &traceProbe{eventName: "__register_chrdev", probeType: kretprobe, programName: "trace_ret__register_chrdev"},
+		DoInitModule:               &traceProbe{eventName: "do_init_module", probeType: kprobe, programName: "trace_do_init_module"},
+		DoInitModuleRet:            &traceProbe{eventName: "do_init_module", probeType: kretprobe, programName: "trace_ret_do_init_module"},
+		LoadElfPhdrs:               &traceProbe{eventName: "load_elf_phdrs", probeType: kprobe, programName: "trace_load_elf_phdrs"},
+		Filldir64:                  &traceProbe{eventName: "filldir64", probeType: kprobe, programName: "trace_filldir64"},
+		UDPSendmsg:                 &traceProbe{eventName: "udp_sendmsg", probeType: kprobe, programName: "trace_udp_sendmsg", netProbe: true},
+		UDPDisconnect:              &traceProbe{eventName: "__udp_disconnect", probeType: kprobe, programName: "trace_udp_disconnect", netProbe: true},
+		UDPDestroySock:             &traceProbe{eventName: "udp_destroy_sock", probeType: kprobe, programName: "trace_udp_destroy_sock", netProbe: true},
+		UDPv6DestroySock:           &traceProbe{eventName: "udpv6_destroy_sock", probeType: kprobe, programName: "trace_udpv6_destroy_sock", netProbe: true},
+		InetSockSetState:           &traceProbe{eventName: "sock:inet_sock_set_state", probeType: rawTracepoint, programName: "tracepoint__inet_sock_set_state", netProbe: true},
+		TCPConnect:                 &traceProbe{eventName: "tcp_connect", probeType: kprobe, programName: "trace_tcp_connect", netProbe: true},
+		ICMPRecv:                   &traceProbe{eventName: "icmp_rcv", probeType: kprobe, programName: "trace_icmp_rcv", netProbe: true},
+		ICMPSend:                   &traceProbe{eventName: "__icmp_send", probeType: kprobe, programName: "trace_icmp_send", netProbe: true},
+		ICMPv6Recv:                 &traceProbe{eventName: "icmpv6_rcv", probeType: kprobe, programName: "trace_icmpv6_rcv", netProbe: true},
+		ICMPv6Send:                 &traceProbe{eventName: "icmp6_send", probeType: kprobe, programName: "trace_icmp6_send", netProbe: true},
+		Pingv4Sendmsg:              &traceProbe{eventName: "ping_v4_sendmsg", probeType: kprobe, programName: "trace_ping_v4_sendmsg"},
+		Pingv6Sendmsg:              &traceProbe{eventName: "ping_v6_sendmsg", probeType: kprobe, programName: "trace_ping_v6_sendmsg"},
+		DefaultTcIngress:           &tcProbe{programName: "tc_ingress", tcAttachPoint: bpf.BPFTcIngress},
```

Comment:
I think these two can be combined into one "probe".
Just use the same fields as we used to have for probes and probeType will be either tcIngress or tcEgress for tc programs.

Reply:
I mean to combine traceProbe and tcProbe to probe, as it was before. Meaning, to adjust the tc probes to fit the existing probe struct.
No need for direction field. Can simply be probeType (the field name we currently have for probe), with either tcIngress or tcEgress for the tc programs.

Reply:
https://github.com/aquasecurity/tracee/pull/1820#issuecomment-1167207980